### ISSUE TITLE
Failed to parse input: Pragma:no-cache

### ISSUE THREAD

**Issue by brianjmurrell** [open-sw/pkg-cacher#35]
I have a node which I have to configure it's `yum.conf` with:

```
http_caching=packages
```

because for some repos it can use a proxy which has bad caching sematics.

_pkg-cacher_ is barfing on having this header set with:

```
Failed to parse input: Pragma:no-cache
```

I think the problem is that _pkg-cacher_ is expecting a space after the `:` in the HTTP headers, but that is not a requirement according to https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html where it says:

> The field value MAY be preceded by any amount of LWS, though a single SP is preferred

(emphasis from original)

**Comment by brianjmurrell** [comment]
Indeed, changing `\s+` to `\s*` on the affected pattern seems to have resolved the issue.  All of the header matching patterns need similar fixing though.

**Comment by jkugler** [comment]
I would be interested to know what "any amount" is intended to mean.  I could see arguments for both "one or more" and "zero or more." Although this seems to imply zero or more is acceptable:
```
       message-header = field-name ":" [ field-value ]
       field-name     = token
       field-value    = *( field-content | LWS )
       field-content  = <the OCTETs making up the field-value
                        and consisting of either *TEXT or combinations
                        of token, separators, and quoted-string>
```

**Comment by brianjmurrell** [comment]
I would say "any amount" means 0 or more.  A.k.a: `/\S*/`.

**Comment by jkugler** [comment]
Agreed.
